src/linenoise_stubs.c

```diff
@@ -57,29 +58,39 @@ static char *hints_bridge(const char *buf, int *color, int *bold)
   if (cb_result == Val_none) {
     CAMLreturnT(char *,NULL);
   } else {
-    char *msg = caml_strdup(String_val(Field(Field(cb_result, 0), 0)));
+    char* msg = caml_stat_strdup(String_val(Field(Field(cb_result, 0), 0)));
```

Comment:
`caml_stat_strdup` was first introduced in ocaml/ocaml#71. We might want to bump the lower bound requirement to OCaml 4.06.